spec/requests/doorkeeper/authorization_spec.rb

```diff
@@ -0,0 +1,48 @@
+require "rails_helper"
+
+RSpec.describe "GET /oauth/authorize" do
+  include Warden::Test::Helpers
+
+  let!(:doorkeeper_app) { create :doorkeeper_application, redirect_uri: "https://localhost:34343/auth/callback" }
+  let!(:resource_owner) { create :user }
+
+  before do
+    Warden.test_mode!
+    login_as(resource_owner, scope: :user)
+  end
+
+  def authorize_url
+    oauth_authorization_url(
+      client_id: doorkeeper_app.uid,
+      redirect_uri: doorkeeper_app.redirect_uri,
+      response_type: "code",
+      state: "c987b113ab6ea39c642f712ba587e8975c05d4fa7bd36fd7"
```

Comment:
The ```client_id, redirect_uri and response_type``` must be like that for the code path we are testing to be taken.
The ```state``` can be anything afaik - I'll update it to an obvious test value